src/ecwrapper.cpp

```diff
@@ -177,8 +179,14 @@ bool CECKey::Recover(const uint256 &hash, const unsigned char *p64, int rec)
     if (rec<0 || rec>=3)
         return false;
     ECDSA_SIG *sig = ECDSA_SIG_new();
-    BN_bin2bn(&p64[0],  32, sig->r);
-    BN_bin2bn(&p64[32], 32, sig->s);
+    BIGNUM *sig_r, *sig_s;
```

Comment:
``ECDSA_SIG_new()`` can also return NULL on allocation failure.

Alternatively, do this:
```
ECDSA_SIG *sig = ECDSA_SIG_new();
BIGNUM *sig_r = BN_bin2bn(&p64[0],  32, nullptr);
BIGNUM *sig_s = BN_BN_bin2bn(&p64[0],  32, nullptr);
assert(sig && sig_r && sig_s);
bool ret = ECDSA_SIG_set0(sig, sig_r, sig_s);
assert(ret);
```

rather than trying to recover.

(But wait, is this code even used? I thought signatures with recovery were not used.)